packages/gatsby/scripts/output-api-file.js

```diff
@@ -0,0 +1,48 @@
+const path = require('path')
+const documentation = require('documentation')
+const fs = require('fs-extra')
+
+const OUTPUT_FILE_NAME = `apis.json`
+
+async function outputFile() {
+  const apis = await Promise.all([
+    path.join('cache-dir', 'api-ssr-docs.js'),
+    path.join('src', 'utils', 'api-browser-docs.js'),
+    path.join('src', 'utils', 'api-node-docs.js')
+  ]
+    .map(filePath => {
+      const resolved = path.resolve(filePath)
+      const [,api] = path.basename(filePath).split('-')
+      return documentation.build(resolved, {
+        shallow: true
+      })
+        .then(contents => {
+          return [
+            contents,
+            api
+          ]
+        })
+    })
+  )
+
+  const output = apis.reduce((merged, [output, api]) => {
+    merged[api] = output.reduce((mergedOutput, doc) => {
+      if (doc.namespace.startsWith('.')) {
```

Comment:
I can't really figure out what this does - worth adding small note why this is here?

Reply:
Yeah -- this should block, good catch!

It's because of this line in gatsby-ssr.js, it's captured as an item in the output when it's not really an API.

```
/**
 * Object containing options defined in `gatsby-config.js`
 * @typedef {object} pluginOptions
 */
```

That creates:

```json
{
  "namespace": "pluginOptions"
}
```

whereas every other one is an export so namespace is `.replaceRenderer` for instance.

Reply:
Right, I now do remember I similar thing to our docs (just implemented differently):

https://github.com/gatsbyjs/gatsby/blob/aea3e39d3e5b19e0c905857f8068bc4d351c2b87/www/src/pages/docs/browser-apis.js#L15-L20

(and I didn't add note on what it does and why it's there :S)